src/core/resources/assets/userAssets.ts

```diff
@@ -272,6 +275,38 @@ export async function parseUserAssets({
     }, {});
     parsedAssetsDict[ChainId.mainnet] = newMainnetAssets;
   }
+  if (connectedToHardhatOp) {
```

Comment:
could we add this stuff to line 242 instead?
most of this code is copy pasted and the only difference i see it the provider we use to get balances (?)

```
if (connectedToHardhat || connectedToHardhatOp) {
   const hardhatChainId = connectedToHardhat ? ChainId.hardhat  : ChainId.hardhatOP
    const provider = getProvider({ chainId: hardhatChainId });
    // force checking for ETH if connected to hardhat
    const mainnetAssets = parsedAssetsDict[hardhatChainId];
    mainnetAssets[ETH_MAINNET_ASSET.uniqueId] = ETH_MAINNET_ASSET;
    if (process.env.IS_TESTING === 'true') {
      mainnetAssets[USDC_MAINNET_ASSET.uniqueId] = USDC_MAINNET_ASSET;
      mainnetAssets[DAI_MAINNET_ASSET.uniqueId] = DAI_MAINNET_ASSET;
    }
    const mainnetBalanceRequests = Object.values(mainnetAssets).map(
      async (asset) => {
        if (asset.chainId !== hardhatChainId) return asset;
...
```

something like that